src/stream/Middleware.test.ts

```diff
@@ -266,25 +266,26 @@ describe('OrderedHistoryResumer', () => {
     expect(subscription).toHaveBeenNthCalledWith(2, null, history[0]);
   });
 
-  it('flushes when empty history page reached', () => {
-    const sequenceId = 0; // out of reach
+  it('flushes but still in seeking state when empty history page reached', () => {
+    const sequenceId = 1; // out of reach
     const middleware = new OrderedHistoryResumer(`${sequenceId}`, 0);
     const subscription = vi.fn();
     middleware.subscribe(subscription);
 
     let history: Types.Message[] = [
+      createMessage(6),
       createMessage(5),
       createMessage(4),
       createMessage(3),
       createMessage(2),
-      createMessage(1),
     ];
     const page1 = history;
     const page2 = [];
 
     expect(middleware.addHistoricalMessages(shuffle(page1))).toBe(false);
-    expect(middleware.addHistoricalMessages(shuffle(page2))).toBe(true);
+    expect(middleware.addHistoricalMessages(page2)).toBe(true);
```

Comment:
is there a reason to remove the shuffle here?

Reply:
I removed it because `page2` is an empty array so there was nothing to shuffle. Would you like me to put it back?